tests/test_resolver.py

```diff
@@ -119,14 +133,22 @@ def generate_expected_resolved_deps(
 @given(
     user_mapping=user_mapping_strategy(user_defined_mapping),
     installed_deps=dict_subset_strategy(locally_installed_deps),
-    other_deps=st.lists(st.sampled_from(other_deps), unique=True),
+    other_deps=dict_subset_strategy(other_deps),
+    install_deps=st.booleans(),
+)
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=5000,
+    max_examples=50,
```

Comment:
These are needed because of the following:
>I opted to make this into a separate PR from the refactor PR because the battery of hypothesis tests slows down significantly with this change (it now takes ~35 seconds to run tests in test_resolver.py).

So I had to both limit the number of generated examples and increase the time limit so that hypothesis testing does not fail on tests taking too long.

Reply:
I see. I suspect the underlying reason is that anytime we pass `install_deps=True`, we're going to spend ~2s extra simply to create the venv, and this this is fundamentally unfriendly to hypothesis which obviuosly wants to call `resolve_dependencies` _a lot_...

As you say above, `test_resolver` is not _really_ about testing each individual resolver, but rather about the logic of combining resolvers in `resolve_dependencies()`. Maybe we can extend the "test intrumentation"  of `TemporaryPipInstallResolver`: Not only do we want a local PyPI mirror to avoid contacting pypi.org in each test, we also want to somehow reuse/cache/fake the virtualenv in which `TemporaryPipInstallResolver` calls `pip install`, so that we can speed up these tests?

Reply:
> Maybe we can extend the "test intrumentation" of TemporaryPipInstallResolver: Not only do we want a local PyPI mirror to avoid contacting pypi.org in each test, we also want to somehow reuse/cache/fake the virtualenv in which TemporaryPipInstallResolver calls pip install, so that we can speed up these tests?

That would be indeed great. However, I don't have any immediate solution popping in my mind right now. Let me sleep on it until tomorrow.